src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.cs

```diff
@@ -5,114 +5,200 @@
+            }
+
+            ImmutableArray<SymbolResult> GetNamespaceSearchResults(
+                SemanticModel semanticModel,
+                TSimpleNameSyntax simpleName,
+                ImmutableArray<ISymbol> symbols)
+            {
+                // There might be multiple namespaces that this name will resolve successfully in.
+                // Some of them may be 'better' results than others.  For example, say you have
+                //  Y.Z   and Y exists in both X1 and X2
+                // We'll want to order them such that we prefer the namespace that will correctly
+                // bind Z off of Y as well.
+
+                string? rightName = null;
+                var isAttributeName = false;
+                if (syntaxFacts.IsLeftSideOfDot(simpleName))
+                {
+                    var rightSide = syntaxFacts.GetRightSideOfDot(simpleName.Parent);
+                    Contract.ThrowIfNull(rightSide);
+
+                    syntaxFacts.GetNameAndArityOfSimpleName(rightSide, out rightName, out _);
+                    isAttributeName = syntaxFacts.IsAttributeName(rightSide);
+                }
+
+                return symbols
+                    .OfType<INamespaceSymbol>()
+                    .Where(n => !n.IsGlobalNamespace && HasAccessibleTypes(n, semanticModel, cancellationToken))
+                    .Select(n => new SymbolResult(n,
+                        BindsWithoutErrors(n, rightName, isAttributeName) ? NamespaceWithNoErrorsWeight : NamespaceWithErrorsWeight))
+                    .ToImmutableArray();
+            }
         }
 
         private IEnumerable<CodeAction> CreateActions(
-            Document document, SyntaxNode node, SemanticModel semanticModel,
+            Document document,
+            SemanticModel semanticModel,
+            TSimpleNameSyntax simpleName,
+            string name,
             IEnumerable<SymbolResult> proposedContainers)
         {
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var ignoreCase = !syntaxFacts.IsCaseSensitive;
+
             foreach (var symbolResult in proposedContainers)
             {
                 var container = symbolResult.Symbol;
-                var containerName = container.ToMinimalDisplayString(semanticModel, node.SpanStart);
-
-                var name = GetNodeName(document, node);
+                Contract.ThrowIfNull(symbolResult.OriginalSymbol);
+                var containerName = container.ToMinimalDisplayString(semanticModel, simpleName.SpanStart);
 
                 // Actual member name might differ by case.
                 string memberName;
-                if (IgnoreCase)
+                if (ignoreCase)
                 {
                     var member = container.GetMembers(name).FirstOrDefault();
                     memberName = member != null ? member.Name : name;
```

Comment:
not your change, but something seems weird here - parts I don't get
1.  the name given (looks like it comes from `GetNameAndArityOfSimpleName`) might not be the right case?
2.  can we always pull the correctly cased name from the actual symbol?

Reply:
so that's what we're trying to do here.  we're trying to find the actual properly cased symbol name (in the case of vb), but we fallback to the name as the user wrote it in case we don't find that symbol.  (note: it seems weird that we wouldn't find it.  the code is likely being too paranoid, but i'm not comfortable making a change there in this PR :)).

Reply:
yeah the part I didn't get is why we need to fallback to anything.  fine to leave it how it is here